Source/WebCore/css/SelectorCheckerTestFunctions.h

```diff
@@ -469,12 +473,24 @@ ALWAYS_INLINE bool matchesPictureInPicturePseudoClass(const Element& element)
 
 ALWAYS_INLINE bool matchesFutureCuePseudoClass(const Element& element)
 {
-    return is<WebVTTElement>(element) && !downcast<WebVTTElement>(element).isPastNode();
+    if (is<WebVTTElement>(element))
+        return !downcast<WebVTTElement>(element).isPastNode();
```

Comment:
```suggestion
    if (auto* element = dynamicDowncast<WebVTTElement>(element)
        return !element->isPastNode();
```
I think the preferred, modern way of doing this is this, but I could be wrong.

Reply:
Yep, I think @rr-codes is correct here. `is<...>` + `downcast<...>` does two two type checks, but `dynamicDowncast<...>` plus a null check does only one.